### ISSUE TITLE
[java] XML format reports wrong class name for inner classes

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#5802]
**Affects PMD Version:** 7.14.0

**Description:**
For example this violation report:

```xml
<file name="/home/andreas/PMD/source/pmd/pmd-core/src/main/java/net/sourceforge/pmd/cpd/impl/BaseTokenFilter.java">
<violation beginline="147" endline="147" begincolumn="51" endcolumn="103" rule="UnnecessaryWarningSuppression" ruleset="Best Practices" package="net.sourceforge.pmd.cpd.impl" class="RemainingTokensIterator" method="computeNext" externalInfoUrl="https://docs.pmd-code.org/pmd-doc-7.14.0/pmd_rules_java_bestpractices.html#unnecessarywarningsuppression" priority="1">
Unnecessary PMD suppression comment
</violation>
</file>
```

It says, package=net.sourceforge.pmd.cpd.impl which is correct, but class=RemainingTokensIterator is wrong - it should be "BaseTokenFilter.RemainingTokens.RemainingTokensIterator"

This leads to wrong output of maven-pmd-plugin:

```
[WARNING] PMD Failure: net.sourceforge.pmd.cpd.impl.RemainingTokensIterator:147 Rule:UnnecessaryWarningSuppression Priority:1 Unnecessary PMD suppression comment.
```
